### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/clojure/tdlib_json/core.clj [hunk 1]
```diff
@@ -1,48 +1,51 @@
...
+  (.execute @client (jsonify messageMap)))
 
-(defn client-receive
-  []
-  (json-parse (. @client receive)))
+(defn client-receive [timeout]
+  (json-parse (.receive @client timeout)))
 
 (defn client-destroy []
+  (.destroy @client)
```
Comment (MityaSaray) [inline src/clojure/tdlib_json/core.clj]: I think you need to send close message to gracefully shut down

Comment (MityaSaray) [reply src/clojure/tdlib_json/core.clj]: https://core.telegram.org/tdlib/docs/classtd_1_1td__api_1_1close.html

File: src/clojure/tdlib_json/core.clj [hunk 2]
```diff
@@ -1,48 +1,51 @@
...
 
-(defn jsonify [messageMap] (che/generate-string messageMap))
 
-(defn json-parse [string]
-  (che/parse-string string true))
+(defn create-client [path-to-lib verbosity]
+  (TgJsonClient. path-to-lib verbosity))
+
+(defn client-send [messageMap]
+  (.send @client (jsonify messageMap)))
```
Comment (MityaSaray) [inline src/clojure/tdlib_json/core.clj]: Lets remove cheshire in this PR. We would want to change this to always receive ^String.

File: src/clojure/tdlib_json/core.clj [hunk 3]
```diff
@@ -1,48 +1,51 @@
...
 
-(defn client-execute
-  ([messageMap] (. @client execute (jsonify messageMap))))
+(defn client-execute [messageMap]
+  (.execute @client (jsonify messageMap)))
```
Comment (MityaSaray) [inline src/clojure/tdlib_json/core.clj]: Same here, change to ^String

File: src/clojure/tdlib_json/core.clj [hunk 4]
```diff
@@ -1,48 +1,51 @@
...
 
-(defn client-receive
-  []
-  (json-parse (. @client receive)))
+(defn client-receive [timeout]
+  (json-parse (.receive @client timeout)))
```
Comment (MityaSaray) [inline src/clojure/tdlib_json/core.clj]: Lets return string here too.

File: src/clojure/tdlib_json/core.clj [hunk 5]
```diff
@@ -1,48 +1,51 @@
...
+   [cheshire.core :as json]
+   [clojure.core.async :as async])
+  (:import [tdlib_json TgJsonClient]))
+
+
+(defn jsonify [messageMap]
+  (json/generate-string messageMap))
+
+(defn json-parse [string]
+  (json/parse-string string true))
```
Comment (MityaSaray) [inline src/clojure/tdlib_json/core.clj]: And we wont need this

**Review by MityaSaray — CHANGES_REQUESTED** [review]
Lets remove cheshire.

File: src/clojure/tdlib_json/core.clj (hunk 2 above)
Comment (ghost) [reply src/clojure/tdlib_json/core.clj]: I think better use docstrings for this. Manual typehinting doesn't add some performance profit (no reflection warnings here).

**Comment by ghost** [comment]
close #5

File: src/clojure/tdlib_json/core.clj [hunk 6]
```diff
@@ -1,48 +1,43 @@
...
 
-(defn client-receive
-  []
-  (json-parse (. @client receive)))
+(defn client-receive [timeout]
+  (.receive ^TgJsonClient @client timeout))
 
 (defn client-destroy []
+  (client-send "{\"@type\":\"close\"}")
+  (.destroy ^TgJsonClient @client)
```
Comment (MityaSaray) [inline src/clojure/tdlib_json/core.clj:25]: There will be no delay between these two calls. On close client sends updateAuthorizationState with authorizationStateClosed message. After this it is safe to destroy client. I am going to add this.

**Pull request merged**